### PR TITLE
WIP:Libstring fs

### PR DESCRIPTION
**Pull request by Athinanarof**
## What is the problem/goal being addressed?
Libstring base64Encode, sha384, sha256 functions implemented.
I'm adding some comments to get feedback.

## What is the solution to this problem?
Functions implemented and test cases included.

## What are the changes here? How do they solve the problem and what other product impacts do they cause?
*Please include before/after screenshots/gifs if appropriate*

## How are you sure this works/how was this tested?
Test cases included, this is WIP.

## What is the reversion plan if this fails after shipping?
Revert changes

### CODE REVIEW

File: fsharp-backend/tests/testfiles/string.tests [hunk 1]
```diff
@@ -19,6 +19,13 @@ String.endsWith_v0 "a string" "in" = false
 String.endsWith_v0 "a string" "ing" = true
 String.foreach_v1 "a string" (fun x -> x) = "a string"
 String.fromChar_v1 'a' = "a"
+String.base64Encode_v0 "" = ""
+String.base64Encode_v0 "f" = "Zg"
+String.base64Encode_v0 "fo" = "Zm8"
```
Comment (pbiggar) [inline fsharp-backend/tests/testfiles/string.tests:24]: Can you test some of the special characters here?

Thanks!

File: fsharp-backend/tests/testfiles/string.tests [hunk 2]
```diff
@@ -26,6 +26,8 @@ String.base64Encode_v0 "foo" = "Zm9v"
 String.base64Encode_v0 "foob" = "Zm9vYg"
 String.base64Encode_v0 "fooba" = "Zm9vYmE"
 String.base64Encode_v0 "foobar" = "Zm9vYmFy"
+String.sha384_v0 "" = "OLBgp1GsljhM2TJ-sbHjaiH9txEUvgdDTAzHv2P24donTt6_529l-9Ua0vFImLlb"
```
Comment (pbiggar) [inline fsharp-backend/tests/testfiles/string.tests:29]: Are the expected values here generated from OCaml?

**Review by pbiggar — APPROVED** [review]

**Comment by pbiggar** [comment]
Looks great. For speed, I'm just merging as there are. Can you address the comments in a subsequent PR? Thanks!

**Comment by Athinanarof** [comment]
Sure! I'm on it!

**Comment by Athinanarof** [comment]
About this "Are the expected values here generated from OCaml?" the answer is yes. (for some reason I can not reply to the conversation above :) )

**Comment by Athinanarof** [comment]
Ooooh, no, forget it I misunderstood the comment. I will work on that.

**Pull request merged**